### PR TITLE
Fix display issues on the democracy dashboard page

### PR DESCRIPTION
**Pull request by gilles437**
Solves #7304

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -112,5 +112,8 @@
     "babel-core": "^7.0.0-bridge.0",
     "styled-components": "^5.3.1",
     "typescript": "^4.6.3"
+  },
+  "dependencies": {
+    "react-table-scrollbar": "^0.1.11"
```
Comment (jsidorenko) [inline package.json:117]: Are you sure you need this? I don't see any usage of that library in the changes below

File: packages/page-democracy/src/Overview/Referendum.tsx [hunk 2]
```diff
@@ -143,6 +147,18 @@ function Referendum ({ className = '', value: { allAye, allNay, image, imageHash
...
+      <td className='expand'>
+        <CardSummary
+          hover={t<string>('Aye')}
+          progress={{
+            hideValue: true,
+            total: totalCalculated,
+            value: votedAye
+          }}
+        ></CardSummary>
+
```
Comment (jsidorenko) [inline packages/page-democracy/src/Overview/Referendum.tsx:159]:
```suggestion
```

File: packages/page-democracy/src/Overview/ReferendumVote.tsx [hunk 3]
```diff
@@ -10,6 +10,8 @@ import { AddressMini } from '@polkadot/react-components';
 
 interface Props {
   vote: DeriveReferendumVote;
+  className?: string;
+
```
Comment (jsidorenko) [inline packages/page-democracy/src/Overview/ReferendumVote.tsx:14]:
```suggestion
```

File: packages/page-democracy/src/Overview/ReferendumVote.tsx [hunk 4]
```diff
@@ -20,6 +22,7 @@ function voteLabel ({ conviction }: Vote, isDelegating: boolean): string {
 
 function ReferendumVote ({ vote: { accountId, balance, isDelegating, vote } }: Props): React.ReactElement<Props> {
   return (
+
```
Comment (jsidorenko) [inline packages/page-democracy/src/Overview/ReferendumVote.tsx:25]:
```suggestion
```

File: packages/page-democracy/src/Overview/Referendums.tsx [hunk 5]
```diff
@@ -42,6 +43,7 @@ function Referendums ({ className = '', referendums }: Props): React.ReactElemen
         />
       ))}
     </Table>
+
```
Comment (jsidorenko) [inline packages/page-democracy/src/Overview/Referendums.tsx:46]:
```suggestion
```

File: package.json (hunk 1 above)
Comment (gilles437) [reply package.json:117]: not needed anymore, I use css style only

File: packages/page-democracy/src/Overview/Referendum.tsx [hunk 6]
```diff
@@ -143,6 +147,18 @@ function Referendum ({ className = '', value: { allAye, allNay, image, imageHash
...
       </td>
+      <td className='expand'>
+        <CardSummary
+          hover={t<string>('Aye')}
+          progress={{
+            hideValue: true,
+            total: totalCalculated,
+            value: votedAye
+          }}
+        ></CardSummary>
```
Comment (jsidorenko) [inline packages/page-democracy/src/Overview/Referendum.tsx:158]:
```suggestion
        <CardSummary
          hover={t<string>('Aye')}
          progress={{
            hideValue: true,
            total: totalCalculated,
            value: votedAye
          }}
        />
```

File: packages/page-democracy/src/Overview/Referendum.tsx [hunk 7]
```diff
@@ -95,7 +96,12 @@ function Referendum ({ className = '', value: { allAye, allNay, image, imageHash
         proposal={image?.proposal}
       />
       <td className='number together media--1200'>
-        <BlockToTime value={remainBlock} />
+        <Badge help={t<string>('remaining')} />
+
+        <BlockToTime
+          value={remainBlock}
+
+        />
```
Comment (jsidorenko) [inline packages/page-democracy/src/Overview/Referendum.tsx:104]:
```suggestion
        <Badge help={t<string>('remaining')} />
        <BlockToTime value={remainBlock} />
```

File: package.json (hunk 1 above)
Comment (jsidorenko) [reply package.json:117]: Pls remove then. And from the yarn.lock as well

Comment (gilles437) [reply package.json:117]: yes, removed

**Comment by jsidorenko** [comment]
LGTM!

**Comment by jacogr** [comment]
Will give the code a run-through as soon as I can focus properly, sorry for the delay. Some small comments while using it:

image

1. The above could use alignment, the graph at the bottom doesn't quite look right. Vertical centering should do the trick.
2. The 100% should be fully black - this could point to a bug in that component itself where 100% is not handled the way it should be...

File: packages/page-democracy/src/Overview/Proposal.tsx [hunk 8]
```diff
@@ -86,4 +96,16 @@ export default React.memo(styled(Proposal)`
       margin-bottom: 4px;
     }
   }
+  
+  div.endorsementsTable {
+
```
Comment (jacogr) [inline packages/page-democracy/src/Overview/Proposal.tsx:101]:
```suggestion
```

File: packages/page-democracy/src/Overview/Proposal.tsx [hunk 9]
```diff
@@ -86,4 +96,16 @@ export default React.memo(styled(Proposal)`
...
+    
+    overflow-y: scroll;
+    display: block;
+    min-height: 50px;
+    max-height: 200px;
+    overflow-x: hidden;
+  }
+  
+
+
```
Comment (jacogr) [inline packages/page-democracy/src/Overview/Proposal.tsx:110]:
```suggestion
```

File: packages/page-democracy/src/Overview/Proposals.tsx [hunk 10]
```diff
@@ -33,11 +33,14 @@ function Proposals ({ className }: Props): React.ReactElement<Props> {
       className={className}
       empty={proposals && t<string>('No active proposals')}
       header={headerRef.current}
+
```
Comment (jacogr) [inline packages/page-democracy/src/Overview/Proposals.tsx:36]:
```suggestion
```

File: packages/page-democracy/src/Overview/Proposals.tsx [hunk 11]
```diff
@@ -33,11 +33,14 @@ function Proposals ({ className }: Props): React.ReactElement<Props> {
...
     >
       {proposals?.map((proposal): React.ReactNode => (
         <ProposalDisplay
+          help={t<string>('launch period')}
           key={proposal.index.toString()}
           value={proposal}
+
```
Comment (jacogr) [inline packages/page-democracy/src/Overview/Proposals.tsx:43]:
```suggestion
```

File: packages/page-democracy/src/Overview/Referendum.tsx [hunk 12]
```diff
@@ -95,7 +96,10 @@ function Referendum ({ className = '', value: { allAye, allNay, image, imageHash
         proposal={image?.proposal}
       />
       <td className='number together media--1200'>
-        <BlockToTime value={remainBlock} />
+        <BlockToTime
+          value={remainBlock}
+
+        />
```
Comment (jacogr) [inline packages/page-democracy/src/Overview/Referendum.tsx:102]:
```suggestion
        <BlockToTime value={remainBlock} />
```

Actutally weird, I was convinced the JSX linting rule should pick up both -

- extra blank lines before the closing tag
- not having a single attribute on a single line (the reverse is certainly true)

File: packages/page-democracy/src/Overview/Referendum.tsx [hunk 13]
```diff
@@ -106,9 +110,7 @@ function Referendum ({ className = '', value: { allAye, allNay, image, imageHash
         {percentages && (
           <>
             <div>{percentages.turnout}</div>
-            {percentages.aye && (
-              <div>{t<string>('{{percentage}} aye', { replace: { percentage: percentages.aye } })}</div>
-            )}
+
```
Comment (jacogr) [inline packages/page-democracy/src/Overview/Referendum.tsx:113]:
```suggestion
```

File: packages/page-democracy/src/Overview/Referendum.tsx [hunk 14]
```diff
@@ -143,6 +145,18 @@ function Referendum ({ className = '', value: { allAye, allNay, image, imageHash
           votes={allNay}
         />
       </td>
+      <td className='class mini media--1000'>
+
```
Comment (jacogr) [inline packages/page-democracy/src/Overview/Referendum.tsx:149]:
```suggestion
```

File: packages/page-democracy/src/Overview/Referendum.tsx [hunk 15]
```diff
@@ -143,6 +145,18 @@ function Referendum ({ className = '', value: { allAye, allNay, image, imageHash
...
+        <CardSummary
+          progress={{
```
Comment (jacogr) [inline packages/page-democracy/src/Overview/Referendum.tsx:151]: I don't believe we should be using a `CardSummary` here at all. The `<Progress ... />` component itself does all the work. (The summary is only a wrapper around it)

File: packages/page-democracy/src/Overview/Referendum.tsx [hunk 16]
```diff
@@ -143,6 +145,18 @@ function Referendum ({ className = '', value: { allAye, allNay, image, imageHash
           votes={allNay}
         />
       </td>
+      <td className='class mini media--1000'>
```
Comment (jacogr) [inline packages/page-democracy/src/Overview/Referendum.tsx:148]:
```suggestion
      <td className='mini media--1000'>
```

class?

File: packages/page-democracy/src/Overview/Referendum.tsx [hunk 17]
```diff
@@ -179,4 +193,27 @@ export default React.memo(styled(Referendum)`
...
+}
+
+.progress-centered-element {
+  margin: 0;
+  position: absolute;
+  top: 50%;
+  transform: translateY(-50%);
+  width: 250;
+
+}
```
Comment (jacogr) [inline packages/page-democracy/src/Overview/Referendum.tsx:216]: I'm convinced this is due to the progress component, at least in this case really don't think we should be using `*` rules unless absolutely needed.

File: packages/page-democracy/src/Overview/ReferendumVotes.tsx [hunk 18]
```diff
@@ -61,14 +61,34 @@ function ReferendumVotes ({ change, className, count, isAye, isWinning, total, v
...
-        <ReferendumVote
-          key={vote.accountId.toString()}
-          vote={vote}
-        />
-      )}
+      <div className='votersTable'>
+        <Table
+          empty={votes && t<string>('No voters')}
+
+        >
```
Comment (jacogr) [inline packages/page-democracy/src/Overview/ReferendumVotes.tsx:68]:
```suggestion
        <Table empty={votes && t<string>('No voters')}>
```

File: packages/page-democracy/src/Overview/ReferendumVotes.tsx [hunk 19]
```diff
@@ -61,14 +61,34 @@ function ReferendumVotes ({ change, className, count, isAye, isWinning, total, v
...
-export default React.memo(ReferendumVotes);
+export default React.memo(styled(ReferendumVotes)`
+div.votersTable {
+  
+  overflow-y: scroll;
+  display: block;
+  min-height: 50px;
+  max-height: 200px;
+  overflow-x: hidden;
+}
```
Comment (jacogr) [inline packages/page-democracy/src/Overview/ReferendumVotes.tsx:93]:
```suggestion
  div.votersTable {
    overflow-y: scroll;
    display: block;
    min-height: 50px;
    max-height: 200px;
    overflow-x: hidden;
  }
```

File: packages/page-democracy/src/Overview/Referendums.tsx [hunk 20]
```diff
@@ -25,6 +25,7 @@ function Referendums ({ className = '', referendums }: Props): React.ReactElemen
     [t('turnout'), 'media--1400'],
     [undefined, 'badge'],
     [t('votes'), 'expand'],
+    [t('aye percentage'), 'media--1000'],
```
Comment (jacogr) [inline packages/page-democracy/src/Overview/Referendums.tsx:28]:
```suggestion
    [undefined, 'media--1000'],
```

The reason here - the label is much wider than the actual data in the column, so it just adds extra whitespace. Additionally it doesn't actually add any information that the graph doesn't convey.

File: packages/react-components/src/CallExpander.tsx [hunk 21]
```diff
@@ -30,6 +30,8 @@ function CallExpander ({ children, className = '', labelHash, value, withHash }:
 
   const { meta, method, section } = call;
 
+  console.log('meta', meta, 'method', method, 'section', section);
+
```
Comment (jacogr) [inline packages/react-components/src/CallExpander.tsx:34]:
```suggestion
```

File: packages/react-components/src/Expander.tsx [hunk 22]
```diff
@@ -85,6 +85,8 @@ function Expander ({ children, className = '', help, helpIcon, isOpen, isPadded,
     [children, renderChildren]
   );
 
+  console.log('summaryHead', summaryHead);
+
```
Comment (jacogr) [inline packages/react-components/src/Expander.tsx:89]:
```suggestion
```

File: packages/react-components/src/TxButton.tsx [hunk 23]
```diff
@@ -96,7 +96,7 @@ function TxButton ({ accountId, className = '', extrinsic: propsExtrinsic, icon,
   return (
     <Button
       className={className}
-      icon={icon || 'check'}
+      icon={icon || 'check-to-slot'}
```
Comment (jacogr) [inline packages/react-components/src/TxButton.tsx:99]:
```suggestion
      icon={icon || 'check'}
```

This is the default value.

File: packages/react-components/src/LinkExternal.tsx [hunk 24]
```diff
@@ -52,7 +52,7 @@ function genLinks (systemChain: string, { data, hash, isLogo, isSidebar, type }:
                 src={logo}
               />
             )
-            : name
+            : 'Show in ' + name
```
Comment (jacogr) [inline packages/react-components/src/LinkExternal.tsx:55]: This should be translatable. Additionally, should be reverted - if you look at the explorer view for a block, it is all-over the page now.

```suggestion
            : name
```

**Review by jacogr — COMMENTED** [review]
Some small code-styling comments and then some functional as well.

There is still some hiccups with the progress display, they probably will be easier to fix with `<Progress .../>` itself

File: packages/page-democracy/src/Overview/Referendums.tsx [hunk 25]
```diff
@@ -25,6 +25,7 @@ function Referendums ({ className = '', referendums }: Props): React.ReactElemen
     [t('turnout'), 'media--1400'],
     [undefined, 'badge'],
     [t('votes'), 'expand'],
+    [undefined, 'media--1000'],
```
Comment (jacogr) [inline packages/page-democracy/src/Overview/Referendums.tsx:28]:
```suggestion
    [t<string>('support'), 'media--1000'],
```

File: packages/page-democracy/src/Overview/Referendum.tsx [hunk 26]
```diff
@@ -143,6 +142,12 @@ function Referendum ({ className = '', value: { allAye, allNay, image, imageHash
           votes={allNay}
         />
       </td>
+      <td className='mini media--1000'>
```
Comment (jacogr) [inline packages/page-democracy/src/Overview/Referendum.tsx:145]:
```suggestion
      <td className='mini media--1000 middle'>
```

**Review by jacogr — COMMENTED** [review]
With these 2 it seems to look ok,

image

Feel free to use/ignore

**Review by jacogr — APPROVED** [review]

**Review by jacogr — APPROVED** [review]
Thank you.

**Pull request merged**